rb_link/src/server/mod.rs

```diff
@@ -0,0 +1,151 @@
+use crate::config::*;
+use serde::{Deserialize, Serialize};
+use std::collections::{HashMap, VecDeque};
+use std::io::{Read, Write};
+use std::os::unix::net::{UnixListener, UnixStream};
+use std::sync::{Arc, Mutex};
+use std::thread::{self, JoinHandle};
+use std::{fs, usize};
+
+mod getter;
+pub use getter::*;
+
+#[derive(Serialize, Deserialize)]
+pub enum GetPutMessage {
+    Get(u32),
+    Put(B2RMessage),
+}
+
+#[derive(Serialize, Deserialize)]
+struct R2BMessage {
+    id: u32,
+    message: Vec<u8>,
+}
+
+/// Message from Bluesim:
+/// - id: ID of the probe that sent the message
+/// - cycles: Clock cycles when the message was sent
+/// - message: Sent message, where message.len() == ceil(put_t_width/8). put_t_width is the width of put_t defined in your BSV code.
+#[derive(Serialize, Deserialize)]
+pub struct B2RMessage {
+    pub id: u32,
+    pub cycles: u32,
+    pub message: Vec<u8>,
+}
+
+/// A server for interacting with Bluesim.
+/// Cache bidirectional data and send data upon receiving requests.
+pub struct B2RServer {
+    socket_path: String,
+    b2r_cache: Arc<Mutex<HashMap<u32, VecDeque<B2RMessage>>>>,
```

Comment:
If the simulation run a long time, will this buffer overflow or lead to OOM?

Reply:
The getter poped from the cache, i think the message shouldn't wait too long to be processed in debug program.
So I think it won't lead to OOM.